### DISCUSSION TITLE
Pass data id to trame

### DISCUSSION THREAD

**Discussion by chriswallis** [Kitware/trame#337] [Q&A]
Hi there,

We would like to pass an id to trame to allow us to pull specific vtu files from our API, similar to what is described [here](https://github.com/Kitware/trame/discussions/148).

I see some examples in the [discussion](https://github.com/Kitware/trame/discussions/83) about using `trame.ClientStateChange` to achieve this from the URL, but I don't think that will fire on the initial page load.

Is there a way to access the initial web request, or is there a better way to do this?

Thank you

**Comment by jourdain** [comment]
The solution was provided in the previous topic. But you can leverage the launcher to pass that info as an argument when you start the process in a multi-user setup.
